src/makeCalculator.js

```diff
@@ -37,7 +37,42 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    operate(method, number) {
+      method.call(this, number);
+
+      return this;
+    },
+    add(number) {
+      this.result = this.result + number;
+
+      return this;
+    },
+    subtract(number) {
+      this.result = this.result - number;
+
+      return this;
+    },
+    multiply(number) {
+      this.result = this.result * number;
+
+      return this;
+    },
+    divide(number) {
+      this.result = this.result / number;
+
+      return this;
+    },
+    reset() {
+      this.result = this.result * 0;
```

Comment:
```suggestion
      this.result *= 0;
```
we can simplify all functions like that